test/miner_payment_v2_txn_SUITE.erl

```diff
@@ -137,3 +140,64 @@ basic_test(Config) ->
                        Miners),
 
     ok.
+
+zero_amt_test(Config) ->
+    Miners = ?config(miners, Config),
+    _ConsensusMiners = ?config(consensus_miners, Config),
+    [Payer | Payees] = Miners,
+    PayerAddr = ct_rpc:call(Payer, blockchain_swarm, pubkey_bin, []),
+
+    PayeeAddrs = lists:foldl(fun(Payee, Acc) ->
+                                     PayeeAddr = ct_rpc:call(Payee, blockchain_swarm, pubkey_bin, []),
+                                     [PayeeAddr | Acc]
+                             end,
+                             [],
+                             Payees),
+
+    ct:pal("PayeeAddrs: ~p", [PayeeAddrs]),
+
+    %% check initial balances
+    5000 = miner_ct_utils:get_balance(Payer, PayerAddr),
+    ok = lists:foreach(fun({Payee, PayeeAddr}) ->
+                               5000 = miner_ct_utils:get_balance(Payee, PayeeAddr)
+                       end,
+                       lists:zip(Payees, PayeeAddrs)),
+
+    Chain = ct_rpc:call(Payer, blockchain_worker, blockchain, []),
+    Ledger = ct_rpc:call(Payer, blockchain, ledger, [Chain]),
+    {ok, Height} = ct_rpc:call(Payer, blockchain, height, [Chain]),
+
+    {ok, Fee} = ct_rpc:call(Payer, blockchain_ledger_v1, transaction_fee, [Ledger]),
+    ct:pal("Fee: ~p", [Fee]),
+
+    %% send 0 amount
+    PayeeAmount = 0,
+    Payments = [blockchain_payment_v2:new(P, PayeeAmount) || P <- PayeeAddrs],
+    Txn = ct_rpc:call(Payer, blockchain_txn_payment_v2, new, [PayerAddr, Payments, 1, Fee]),
+
+    {ok, _Pubkey, SigFun, _ECDHFun} = ct_rpc:call(Payer, blockchain_swarm, keys, []),
+
+    SignedTxn = ct_rpc:call(Payer, blockchain_txn_payment_v2, sign, [Txn, SigFun]),
+    ct:pal("SignedTxn: ~p", [SignedTxn]),
+
+    {[], [SignedTxn]} = ct_rpc:call(Payer, blockchain_txn, validate, [[SignedTxn], Chain]),
```

Comment:
Kinda a nit but maybe better to do something like ?assertEqual(false, TXN_TYPE:is_valid(Txn, Chain)) ?
Make the validation check more explicit and obvious

Reply:
Certainly, added d149bca